org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/JavaDependsOnScalaBothAreOkTest.scala

```diff
@@ -0,0 +1,67 @@
+package org.scalaide.core
+package sbtbuilder
+
+import org.junit.BeforeClass
+import org.scalaide.core.testsetup.SDTTestUtils
+import org.junit.Assert._
+import org.junit.Test
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.resources.IMarker
+
+object JavaDependsOnScalaBothAreOkTest extends testsetup.TestProjectSetup("javaDependsOnScalaBothAreOk") {
+  @BeforeClass def setup(): Unit = {
+    SDTTestUtils.enableAutoBuild(false)
+  }
+}
+
+class JavaDependsOnScalaBothAreOkTest {
+  import JavaDependsOnScalaBothAreOkTest._
+
+  @Test def shouldCreateProjectFromOkJavaAndSpoilAndFixIt(): Unit = {
+    val TypeBMustImplementFoo = 1
+    cleanProject()
+    import SDTTestUtils._
+
+    val aClass = compilationUnit("main/A.scala")
+    val bClass = compilationUnit("main/B.java")
+    def rebuildAndCollectProblems(): List[IMarker] = {
+      project.underlying.build(IncrementalProjectBuilder.INCREMENTAL_BUILD, new NullProgressMonitor)
+      getProblemMarkers(aClass, bClass)
+    }
+
+    step("first compile scala and dependent java classes") {
+      val problems = rebuildAndCollectProblems()
+      assertTrue("No build error expected, got: " + markersMessages(problems), problems.isEmpty)
+    }
+
+    step("then extends java B with abstract A") {
+      changeContentOfFile(bClass.getResource().getAdapter(classOf[IFile]).asInstanceOf[IFile], changed_main_B)
```

Comment:
I just saw, there already exist `EclipseUtils.adaptTo`, which is doing what you want.